tests/unit/test_vcs.py

```diff
@@ -396,30 +397,101 @@ def test_subversion__get_vcs_version():
-                                             expected_version):
+def test_subversion__call_vcs_version_patched(mock_run_command, svn_output,
+                                              expected_version):
     """
-    Test Subversion.get_vcs_version() against patched output.
+    Test Subversion.call_vcs_version() against patched output.
     """
     mock_run_command.return_value = svn_output
-    version = Subversion().get_vcs_version()
+    version = Subversion().call_vcs_version()
     assert version == expected_version
 
 
 @patch('pip._internal.vcs.subversion.Subversion.run_command')
-def test_subversion__get_vcs_version_svn_not_installed(mock_run_command):
+def test_subversion__call_vcs_version_svn_not_installed(mock_run_command):
     """
-    Test Subversion.get_vcs_version() when svn is not installed.
+    Test Subversion.call_vcs_version() when svn is not installed.
     """
     mock_run_command.side_effect = BadCommand
     with pytest.raises(BadCommand):
-        Subversion().get_vcs_version()
+        Subversion().call_vcs_version()
+
+
+@pytest.mark.parametrize('version', [
+    (),
+    (1,),
+    (1, 8),
+    (1, 8, 0),
+])
+def test_subversion__get_vcs_version_cached(version):
+    """
+    Test Subversion.get_vcs_version() with previously cached result.
+    """
+    sub = Subversion()
+    sub._vcs_version = version
+    assert sub.get_vcs_version() == version
+
+
+@pytest.mark.parametrize('vcs_version', [
+    (),
+    (1, 7),
+    (1, 8, 0),
+])
+@patch('pip._internal.vcs.subversion.Subversion.call_vcs_version')
+def test_subversion__get_vcs_version_call_vcs(mock_call_vcs, vcs_version):
+    """
+    Test Subversion.get_vcs_version() with mocked output from
+    call_vcs_version().
+    """
+    mock_call_vcs.return_value = vcs_version
+    sub = Subversion()
+    assert sub.get_vcs_version() == vcs_version
+
+    # Check that the version information is cached.
+    assert sub._vcs_version == vcs_version
+
+
+@pytest.mark.parametrize('use_interactive,vcs_version,expected_options', [
+    (False, (), ['--non-interactive']),
+    (False, (1, 7, 0), ['--non-interactive']),
+    (False, (1, 8, 0), ['--non-interactive']),
+    (True, (), []),
+    (True, (1, 7, 0), []),
+    (True, (1, 8, 0), ['--force-interactive']),
+])
+def test_subversion__get_vcs_global_options(use_interactive, vcs_version,
+                                            expected_options):
+    """
+    Test Subversion.get_vcs_global_options().
+    """
+    sub = Subversion(use_interactive=use_interactive)
+    sub._vcs_version = vcs_version
+    assert sub.get_vcs_global_options() == expected_options
+
+
+@pytest.mark.parametrize('isatty_return,use_interactive,expected', [
+    (False, None, False),
+    (False, False, False),
+    (False, True, True),
+    (True, None, True),
+    (True, False, False),
+    (True, True, True),
+])
+@patch('sys.stdin.isatty')
+def test_subversion__constructor_use_interactive(mock_stdin, isatty_return,
```

Comment:
A few things here:

* I would try to order the tests in the same order that the methods appear in the class.
* I think it would be clearer to name this `test_subversion__init...` instead of `test_subversion__constructor...`, to match the method name being tested.
* To avoid large indents with long function names, the style more commonly being used in the code base is this one--

```
def test_subversion__constructor_use_interactive(
    mock_stdin, isatty_return, use_interactive, expected,
):
    """
    ...
```